### PR TITLE
Support vulkan on linux

### PR DESCRIPTION
**Pull request by kd-11**
This PR enables building rpcs3 on linux with working vulkan support. List of changes includes:
 - Fix build scripts to allow building on linux with working vulkan support
 - Drop unused stuff from the main rpcs3 project
 - Reworks cmake scripts a bit to make cmake behave in a more predictable manner including deleting autogenerated Qt moc files when running 'make clean'
- Fix a few vulkan API spec violations. Fixes a free-during-use on a leaky handle when the debug overlay is enabled. Also clips clear rects to always fit within the current render target.

~~As noted in the title, this PR is WIP~~

### CODE REVIEW

**Comment by AniLeo** [comment]
Solves #1752 , #2186 , #2255 , #2479 (Controller issue will remain but there's #2678 for it)

**Comment by kd-11** [comment]
Looks like vulkan.7z needs to be repackaged for appveyor. Had to update vulkan submodules due to a gcc conflict between <math.h> and <cmath> that was fixed upstream.

**Comment by mrcmunir** [comment]
Good up to changes 

 in vulkan mesa3d driver does not support the VK_FORMAT_D24_UNORM_S8_UINT  !? 

Returned St13runtime_error thrown: No hardware support for z24s8 I reported this year ago in mesa3d dev's but not changed yet.

If disable this check will run vulkan backend but some glitches / inverted colour like this 

captura de 2017-06-07 20-47-20
captura de 2017-06-07 20-47-04

**Comment by kd-11** [comment]
Interesting on the d24s8 issue. I did not notice it. Been having some X hangups with radv making it harder to fix everything, but I'm ironing out the remaining issues.
BTW radeon hardware does not support 24-bit depth, instead jumping from 16-bit to 32-bit. A minor workaround should fix it.

File: rpcs3/rpcs3qt.vcxproj.user [hunk 1]
```diff
@@ -0,0 +1,4 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<Project ToolsVersion="14.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <PropertyGroup />
+</Project>
```
Comment (Zangetsu38) [inline rpcs3/rpcs3qt.vcxproj.user]: this file already exist, it is rpcs3.vcxproj.user ^^, i fix this problem in my pr

Comment (kd-11) [inline rpcs3/rpcs3qt.vcxproj.user]: Added this file by mistake. This commit should only be about the leaking direct_fbo handle

Comment (Zangetsu38) [inline rpcs3/rpcs3qt.vcxproj.user]: oh ok :)

**Pull request merged**